### PR TITLE
update samtools/faidx

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

This PR updates the recipe so it can extract sequences as well. 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by peterpru — APPROVED** [review]
Looks good to me!

File: modules/nf-core/samtools/faidx/main.nf [hunk 1]
```diff
@@ -9,11 +9,13 @@ process SAMTOOLS_FAIDX {
 
     input:
     tuple val(meta), path(fasta)
+    tuple val(meta2), path(fai)
```
Comment (maxulysse) [inline modules/nf-core/samtools/faidx/main.nf:12]: Why would we need fai as input when it's what we want to output?

File: modules/nf-core/samtools/faidx/main.nf [hunk 2]
```diff
@@ -9,11 +9,13 @@ process SAMTOOLS_FAIDX {
...
 
     output:
-    tuple val(meta), path ("*.fai"), emit: fai
-    tuple val(meta), path ("*.gzi"), emit: gzi, optional: true
-    path "versions.yml"            , emit: versions
+    tuple val(meta), path ("*.{fa,fasta}") , emit: fa , optional: true
```
Comment (maxulysse) [inline modules/nf-core/samtools/faidx/main.nf:15]: Why outputting fa?
any difference between the input?

File: modules/nf-core/samtools/faidx/main.nf (hunk 1 above)
Comment (ramprasadn) [reply modules/nf-core/samtools/faidx/main.nf:12]: When you give it both fasta and fai, you can extract specific sequences as well.

File: modules/nf-core/samtools/faidx/main.nf (hunk 2 above)
Comment (ramprasadn) [reply modules/nf-core/samtools/faidx/main.nf:15]: Yeah, when you extract sequences using faidx, output is written in fasta format

**Review by maxulysse — APPROVED** [review]

File: modules/nf-core/samtools/faidx/main.nf (hunk 2 above)
Comment (yfarjoun) [reply modules/nf-core/samtools/faidx/main.nf:15]: but...you didn't pass the second argument to samtools...so what is the purpose of this?

**Pull request merged**